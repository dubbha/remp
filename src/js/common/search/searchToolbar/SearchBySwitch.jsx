import React from 'react';
import PropTypes from 'prop-types';
import SearchButton from './SearchButton';

const SearchBySwitch = ({ searchBy, onSearchByChange, searchByParams }) => (
  <div className="searchBySwitch">
    <span className="searchBySwitch__text">Search By</span>
    {
      searchByParams.map(item => (
        <SearchButton
          key={item}
          text={item}
          active={searchBy === item}
          onClickWithText={onSearchByChange}
        />
      ))
    }
  </div>
);

SearchBySwitch.propTypes = {
  searchBy: PropTypes.string.isRequired,
  onSearchByChange: PropTypes.func.isRequired,
  searchByParams: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SearchBySwitch;
