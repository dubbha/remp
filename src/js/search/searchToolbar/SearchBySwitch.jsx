import React from 'react';
import SearchButton from './SearchButton';

const SearchBySwitch = () => (
  <div className="searchBySwitch">
    <span className="searchBySwitch__text">Search By</span>
    <SearchButton
      text="Title"
      active
    />
    <SearchButton
      text="Director"
    />
  </div>
);

export default SearchBySwitch;
