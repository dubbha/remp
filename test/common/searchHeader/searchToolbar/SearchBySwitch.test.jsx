import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBySwitch from 'common/searchHeader/searchToolbar/SearchBySwitch';

describe('SearchSortByLink', () => {
  let props;

  beforeEach(() => {
    props = {
      searchBy: 'title',
      searchByParams: ['title', 'director'],
      onSearchByChange: jest.fn(),
    };
  });

  it('should render successfully', () => {
    const wrapper = shallow(<SearchBySwitch {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should switch the active link according to the selected sort by parameter', () => {
    props.searchBy = 'director';
    const wrapper = shallow(<SearchBySwitch {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
