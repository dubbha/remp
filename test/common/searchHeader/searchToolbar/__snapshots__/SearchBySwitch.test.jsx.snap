// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSortByLink should render successfully 1`] = `
<div
  className="searchBySwitch"
>
  <span
    className="searchBySwitch__text"
  >
    Search By
  </span>
  <SearchButton
    active={true}
    key="title"
    onClickWithText={[Function]}
    size="small"
    text="title"
    type="button"
  />
  <SearchButton
    active={false}
    key="director"
    onClickWithText={[Function]}
    size="small"
    text="director"
    type="button"
  />
</div>
`;

exports[`SearchSortByLink should switch the active link according to the selected sort by parameter 1`] = `
<div
  className="searchBySwitch"
>
  <span
    className="searchBySwitch__text"
  >
    Search By
  </span>
  <SearchButton
    active={false}
    key="title"
    onClickWithText={[Function]}
    size="small"
    text="title"
    type="button"
  />
  <SearchButton
    active={true}
    key="director"
    onClickWithText={[Function]}
    size="small"
    text="director"
    type="button"
  />
</div>
`;
